### PR TITLE
[SPARK-41647][CONNECT] Deduplicate docstrings in pyspark.sql.connect.functions

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to deduplicate docstrings in `pyspark.sql.connect.functions`.

### Why are the changes needed?

For easier maintenance

### Does this PR introduce _any_ user-facing change?

No, dev-only. There're mintor doc changes that mention about remote support in Apache Spark 3.4.

### How was this patch tested?

No test. It has to be manually verified until we resolve SPARK-41653.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
To make sure, I wrote a script and updated the codes. I also manually checked one by one.

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -132,6 +132,9 @@ def lit(col: Any) -> Column:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py:132]: Missing functions in Spark Connect are:

```
window_time
call_udf
days
window
hours
pandas_udf
sequence
bucket
years
session_window
format_number
unwrap_udt
udf
months
```

FYI @zhengruifeng

**Review by itholic — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: python/pyspark/sql/functions.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/functions.py:132]: thank you, 

the `Partition Transformation Functions` is tracked in https://issues.apache.org/jira/browse/SPARK-41503

I will add udf-related function after udf is supported

**Pull request closed without merging**